noxfile.py

```diff
@@ -30,68 +33,76 @@
 
 COVERAGE_PACKAGES = ["pytest-cov", "codecov"]
 IRIS_GITHUB = "https://github.com/scitools/iris.git"
+LOCKFILE_PLATFORM = "linux-64"
 
 
-def _cache_venv(session, fname):
+def _lockfile_path(py_string: str, platform_placeholder: bool = False) -> Path:
     """
-    Cache the nox session environment.
-
-    This consists of saving a hexdigest (sha256) of the associated
-    conda requirements YAML file.
+    Return a constructed lockfile path for the relevant python string e.g ``py38``.
 
-    Parameters
-    ----------
-    session: object
-        A `nox.sessions.Session` object.
-    fname: str
-        Requirements filename that defines this environment.
+    Optionally retain the ``{platform}`` placeholder to support conda-lock's
+    ``--filename-template``.
 
     """
-    python_version = session.python.replace(".", "")
-    with open(fname, "rb") as fi:
-        hexdigest = hashlib.sha256(fi.read()).hexdigest()
+    dir = Path() / "requirements" / "nox.lock"
+    name_template = "{py_string}-{platform}.lock"
+    if platform_placeholder:
+        platform = "{platform}"
+    else:
+        platform = LOCKFILE_PLATFORM
+    lockfile_name = name_template.format(py_string=py_string, platform=platform)
+    return dir / lockfile_name
+
+
+def _session_lockfile(session: nox.sessions.Session) -> Path:
+    """Return the path of the session lockfile."""
+    return _lockfile_path(py_string=f"py{session.python.replace('.', '')}")
+
+
+def _session_cachefile(session: nox.sessions.Session) -> Path:
+    """Return the path of the session lockfile cache."""
     tmp_dir = Path(session.create_tmp())
-    cache = tmp_dir / f"py{python_version}.sha"
-    with open(cache, "w") as fo:
-        fo.write(hexdigest)
+    cache = tmp_dir / _session_lockfile(session).name
+    return cache
+
+
+def _venv_populated(session: nox.sessions.Session) -> bool:
+    """Return True if the Conda venv has been created and the list of packages in the lockfile installed."""
+    return _session_cachefile(session).is_file()
+
+
+def _venv_changed(session: nox.sessions.Session) -> bool:
+    """Return True if the installed session is different to that specified in the lockfile."""
+    result = False
+    if _venv_populated(session):
+        with _session_lockfile(session).open("rb") as lockfile:
```

Comment:
I suppose a simpler alternative here might just be to cache the actual lockfile, rather than a hash of the lockfile?  It would also allow us to do introspection on the differences if we ever wanted to

Reply:
(I know this is the same as I did for iris and was based on the previous caching technique, it's now struck me that this might not be the optimal solution!)

Reply:
Nice, I'll get on that

Reply:
e800e5b7281e00984910268cebecb68f2f735f41